Samples/StorageProviders/StorageProviders/BaseJSONStorageProvider.cs

```diff
@@ -77,7 +86,7 @@ public Task Close()
         public async Task ReadStateAsync(string grainType, GrainReference grainReference, IGrainState grainState)
         {
             if (DataManager == null) throw new ArgumentException("DataManager property not initialized");
-            var entityData = await DataManager.Read(grainState.GetType().Name, grainReference.ToKeyString());
+            var entityData = await DataManager.Read(grainState.State.GetType().Name, grainReference.ToKeyString());
```

Comment:
Wouldn't it be better to use grainType as the collection name? otherwise if you have multiple grain types saving the same kind of state, you'd be piling them on the same collection (with potential ID conflicts)...